src/utils/upload/processUploadV1.js

```diff
@@ -0,0 +1,272 @@
+/* eslint-disable no-param-reassign */
+import _ from 'lodash';
+
+import axios from 'axios';
+
+import { createSample, updateSampleFile } from 'redux/actions/samples';
+
+import fetchAPI from 'utils/http/fetchAPI';
+
+import UploadStatus from 'utils/upload/UploadStatus';
+import loadAndCompressIfNecessary from 'utils/upload/loadAndCompressIfNecessary';
+import { inspectFile, Verdict } from 'utils/upload/fileInspector';
+
+const putInS3 = async (projectUuid, loadedFileData, dispatch, sampleUuid, fileName, metadata) => {
+  const baseUrl = `/v1/projects/${projectUuid}/samples/${sampleUuid}/${fileName}/uploadUrl`;
+
+  const urlParams = new URLSearchParams(metadata);
+  const urlParamsStr = Object.keys(metadata).length ? `?${urlParams}` : '';
+
+  const url = `${baseUrl}${urlParamsStr}`;
+
+  const signedUrl = await fetchAPI(
+    url,
+    {
+      method: 'GET',
+    },
+  );
+
+  return axios.request({
+    method: 'put',
+    url: signedUrl,
+    data: loadedFileData,
+    headers: {
+      'Content-Type': 'application/octet-stream',
+    },
+    onUploadProgress: (progress) => {
+      const percentProgress = Math.round((progress.loaded / progress.total) * 100);
+
+      dispatch(updateSampleFile(sampleUuid, fileName, {
+        upload: {
+          status: UploadStatus.UPLOADING,
+          progress: percentProgress ?? 0,
+        },
+      }));
+    },
+  });
+};
+
+const metadataForFile = (file) => {
+  const metadata = {};
+
+  if (file.name.includes('genes')) {
+    metadata.cellranger_version = 'v2';
+  } else if (file.name.includes('features')) {
+    metadata.cellranger_version = 'v3';
+  }
+
+  return metadata;
+};
+
+const compressAndUploadSingleFile = async (
+  projectUuid, sampleUuid, fileName, file,
+  dispatch, metadata = {},
+) => {
+  let loadedFile = null;
+
+  const filePropertiesToUpdate = {
+    size: file.size,
+    fileObject: file.fileObject,
+  };
+
+  try {
+    loadedFile = await loadAndCompressIfNecessary(file, () => (
+      dispatch(
+        updateSampleFile(
+          sampleUuid,
+          fileName,
+          { upload: { status: UploadStatus.COMPRESSING } },
+        ),
+      )
+    ));
+  } catch (e) {
+    const fileErrorStatus = e === 'aborted' ? UploadStatus.FILE_READ_ABORTED : UploadStatus.FILE_READ_ERROR;
```

Comment:
Shouldn't this be `e.message === 'aborted'`? `e` is an instance of `Error` right?